File: packages/sdk/src/migrations/snapshotMigration0003.ts
import { Snapshot } from '@towns-protocol/proto'

/**
 * Zero out the tips sent and received since we're adding counts and we want them to match
 */
export function snapshotMigration0003(snapshot: Snapshot): Snapshot {
    switch (snapshot.content?.case) {
        case 'userContent': {
            snapshot.content.value.tipsSent = {}
            snapshot.content.value.tipsReceived = {}
            snapshot.content.value.tipsSentCount = {}
            snapshot.content.value.tipsReceivedCount = {}
        }
    }
    return snapshot
}

File: core/node/events/migrations/snapshot_migration_0003.go
package migrations

import (
	. "github.com/towns-protocol/towns/core/node/protocol"
)

// zero out the tips sent and received since we're adding counts and we want them to match
func snapshot_migration_0003(iSnapshot *Snapshot) {
	switch snapshot := iSnapshot.Content.(type) {
	case *Snapshot_UserContent:
		snapshot.UserContent.TipsSent = make(map[string]uint64)
		snapshot.UserContent.TipsReceived = make(map[string]uint64)
		snapshot.UserContent.TipsSentCount = make(map[string]uint64)
		snapshot.UserContent.TipsReceivedCount = make(map[string]uint64)
	}
}
